test/unit/test_add_unit.py

```diff
@@ -0,0 +1,43 @@
+import jubilant
+
+from . import mocks
+
+
+def test_defaults(run: mocks.Run):
+    run.handle(['juju', 'add-unit', 'app1'])
+    juju = jubilant.Juju()
+
+    juju.add_unit('app1')
+
+
+def test_defaults_with_model(run: mocks.Run):
+    run.handle(['juju', 'add-unit', '--model', 'mdl', 'app1'])
+    juju = jubilant.Juju(model='mdl')
+
+    juju.add_unit('app1')
+
+
+def test_all_args(run: mocks.Run):
+    run.handle(
+        [
+            'juju',
+            'add-unit',
+            'app2',
+            '--attach-storage',
+            'stg',
+            '--num-units',
+            '3',
+            '--to',
+            'lxd:25',
+        ]
+    )
+    juju = jubilant.Juju()
+
+    juju.add_unit('app2', attach_storage='stg', num_units=3, to='lxd:25')
```

Comment:
Would I be correct to assume that the run mock requires exactly one call with exact same arguments?

Reply:
>= 1 calls with the exact same arguments, yes. It seems to work well!